lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/classify/ArithmeticClassifier.java

```diff
@@ -0,0 +1,162 @@
+package edu.stanford.futuredata.macrobase.analysis.classify;
+
+import edu.stanford.futuredata.macrobase.datamodel.DataFrame;
+
+import java.lang.Double;
+import org.apache.commons.math3.stat.descriptive.rank.Percentile;
+import org.apache.commons.math3.distribution.NormalDistribution;
+
+/**
+ * Classify rows by high / low values based on the group mean and standard deviation.
+ * Returns a new dataframe with a column representation of the estimated number of outliers
+ * for each group, which can be non-integer.
+ */
+public class ArithmeticClassifier extends CubeClassifier implements ThresholdClassifier {
+    // Parameters
+    private String meanColumnName = "mean";
+    private String stdColumnName = "std";
+    private double percentile = 1.0;
+    private boolean includeHigh = true;
+    private boolean includeLow = true;
+
+    // Calculated values
+    private double lowCutoff;
+    private double highCutoff;
+    private DataFrame output;
+
+    public ArithmeticClassifier(String countColumnName, String meanColumnName,
+                                String stdColumnName) {
+        super(countColumnName);
+        this.meanColumnName = meanColumnName;
+        this.stdColumnName = stdColumnName;
+    }
+
+    @Override
+    public void process(DataFrame input) {
+        double[] means = input.getDoubleColumnByName(meanColumnName);
+        double[] counts = input.getDoubleColumnByName(countColumnName);
+        double[] stds = input.getDoubleColumnByName(stdColumnName);
+        int len = means.length;
+        int numRawMetrics = 0;
+        for (int i = 0; i < len; i++) {
+            numRawMetrics += counts[i];
+        }
+        double[] rawMetrics = new double[numRawMetrics];
+        int cumRawMetrics = 0;
+        for (int i = 0; i < len; i++) {
+            for (int j = cumRawMetrics; j < cumRawMetrics + counts[i]; j++) {
+                rawMetrics[j] = means[i];
+            }
+            cumRawMetrics += counts[i];
+        }
+        lowCutoff = new Percentile().evaluate(rawMetrics, percentile);
+        highCutoff = new Percentile().evaluate(rawMetrics, 100.0 - percentile);
+
+        output = input.copy();
+        double[] resultColumn = new double[len];
+        for (int i = 0; i < len; i++) {
+            double mean = means[i];
+            double std = stds[i];
+            double count = counts[i];
+            double numOutliers = 0.0;
+            if (Double.isNaN(std) || std == 0.0) {
```

Comment:
if there's only one raw event in that group, the standard deviation comes out to NaN, at least when I did aggregation with Pandas.